internal/dotnet/plan.go

```diff
@@ -7,8 +7,37 @@ import (
 
 	"github.com/spf13/afero"
 	"github.com/zeabur/zbpack/internal/utils"
+	"github.com/zeabur/zbpack/pkg/types"
 )
 
+// DetermineFramework is used to determine the Dotnet framework
+func DetermineFramework(entryPoint string, src afero.Fs) (string, error) {
+	fileName := entryPoint + ".csproj"
+	if utils.HasFile(src, fileName) {
+		content, err := afero.ReadFile(src, fileName)
```

Comment:
Check-then-read is a bad pattern since it ignores the possibility that users may remove the file before `afero.ReadFile` and after the `utils.HasFile` determination. How about:

```go
content, err := afero.ReadFile(src, fileName)
if err != nil {
  if errors.Is(err, os.ErrNotExist) {
    return "", errors.New("Unable to determine framework")
  }

  return "", err
}
```

Reply:
@pan93412 I'm checking for error on file read as well.

```
content, err := afero.ReadFile(src, fileName)
if err != nil {
	return "", err	
}
```

Though, I think it might be better to return a file does not exist error, instead of "Unable to determine framework". What do you think?